include/deal.II/base/point.h

```diff
@@ -157,8 +154,23 @@ class Point : public Tensor<1, dim, Number>
    *
    * @note This function can also be used in CUDA device code.
    */
-  DEAL_II_CUDA_HOST_DEV
-  Point(const Number x, const Number y, const Number z);
+  DEAL_II_CUDA_HOST_DEV constexpr Point(const Number x,
+                                        const Number y,
+                                        const Number z);
+
+  /**
+   * Constructor for four dimensional points. This function is only
+   * implemented for <tt>dim==4</tt> since the usage is considered unsafe for
+   * points with <tt>dim!=4</tt> as it would leave some components of the
+   * point coordinates uninitialized (if dim>4) or would not use some
+   * arguments (if dim<4).
+   *
+   * @note This function can also be used in CUDA device code.
+   */
+  DEAL_II_CUDA_HOST_DEV constexpr Point(const Number x,
+                                        const Number y,
+                                        const Number z,
+                                        const Number w);
```

Comment:
Why did you this function?

Reply:
In order to fix compilation of `mapping_cartesian.cc`, I added the array `GeometryInfo<dim>::unit_normal_vector` as it made the code more readable. Then I noticed that `GeometryInfo<dim>` is also defined for `dim == 4`. So then I had to add a constexpr constructor for `Point<4>`. 

This was my decision process. Let me know your suggestions.